python/modem-firmware-1.3+/device_credentials_installer.py

```diff
@@ -699,5 +712,63 @@ def main():
 
     cleanup()
 
+def verify_credentials(sec_tag, ca_cert, client_cert):
+    write_line('AT%CMNG=1,{},0'.format(sec_tag))
+    retval, ca_res = wait_for_prompt(b'OK', b'ERROR', store=b'\r\n')
+    if retval:
+        retval = check_sha(ca_res.decode(), ca_cert, 'CA Cert')
+
+    if not retval:
+        return False
+
+    write_line('AT%CMNG=1,{},1'.format(sec_tag))
+    retval, client_res = wait_for_prompt(b'OK', b'ERROR', store=b'\r\n')
+    if retval:
+        retval = check_sha(client_res.decode(), client_cert, 'Client Cert')
+
+    if not retval:
+        return False
+
+    prv_sha = None
+    write_line('AT%CMNG=1,{},2'.format(sec_tag))
+    retval, prv_res = wait_for_prompt(b'OK', b'ERROR', store=b'\r\n')
+    if retval:
+        prv_sha = parse_sha(prv_res.decode())
+
+    if not prv_sha:
+        print(error_style('Private key not found'))
+        return False
+
+    print(hivis_style('Private Key SHA: {}'.format(prv_sha)))
+    return True
+
+def parse_sha(cmng_result_str):
+    try:
+        return cmng_result_str.split('"')[1]
```

Comment:
This function feels fragile to me

If I understand correctly, you're, on some level, using presence of quotes to detect whether or not the sha was found

I would recommend instead, perhaps:
```py
def parse_at_resp(at_response):
   return at_response.split(',')
```

And then extracting the third argument of that if it exists + some other more direct check of whether the cert requested actually exists in the first place

Reply:
example response:
`%CMNG: 16842753,2,"4F7BAA4897065359990D982286C1186F0AEA3ADC0C992F8939075150A4D14E81"`
the SHA is in quotes.

Reply:
Yeah, I understand -- I'm just saying that that response is comma-separated data, so it might make more sense to someone reading your code if you parse it first by splitting the data by commas

Also I think your current version also detects whether the cert was even present by checking whether or not there was a SHA in the response, (in turn by checking whether `"` is in the response), and that might also throw off a reader